File: game/forest/hurt_flying_start.py
import pygame

from animation import Animation
from audio_helper import AudioHelper
from forest.forest_resources import ForestResources
from forest.hurt_flying_talking import HurtFlyingTalking
from state import State


class HurtFlyingStart(State):
    def process_events(self, events):
        if self.get_frame_index_fast() >= len(ForestResources.catapult_fly):
            return HurtFlyingTalking

        if self.one_shot(2.7, "crash"):
            AudioHelper.play(ForestResources.sfx_hugo_screenklir, self.context.country)

        if self.one_shot(2.7, "uy"):
            AudioHelper.play(ForestResources.speak_catapult_hit, self.context.country)

        return None

    def on_enter(self) -> None:
        super().on_enter()
        AudioHelper.play(ForestResources.speak_catapult_up, self.context.country)

    def render(self, screen):
        screen.blit(Animation.get_frame(ForestResources.catapult_fly, self.get_frame_index_fast()), (0,0))

File: game-web/src/forest/hurtFlyingStart.ts
/** Port of game/forest/hurt_flying_start.py */

import { State, type StateClass } from '../state';
import { Animation } from '../animation';
import { AudioHelper } from '../audioHelper';
import { ForestResources } from './forestResources';
import { HurtFlyingTalking } from './hurtFlyingTalking';
import type { PhoneEvents } from '../phoneEvents';

export class HurtFlyingStart extends State {
  processEvents(_events: PhoneEvents): StateClass | null {
    if (this.getFrameIndexFast() >= ForestResources.catapultFly.length) return HurtFlyingTalking;
    if (this.oneShot(2.7, 'crash')) AudioHelper.play(ForestResources.sfxHugoScreenklir);
    if (this.oneShot(2.7, 'uy'))   AudioHelper.play(ForestResources.speakCatapultHit);
    return null;
  }

  onEnter(): void {
    super.onEnter();
    AudioHelper.play(ForestResources.speakCatapultUp);
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    ctx.drawImage(Animation.getFrame(ForestResources.catapultFly, this.getFrameIndexFast()), 0, 0);
  }
}
